apps/framework-cli/src/framework/core/plan_validator.rs

```diff
@@ -10,11 +10,72 @@ pub enum ValidationError {
 
     #[error("Table validation failed: {0}")]
     TableValidation(String),
+
+    #[error("Cluster validation failed: {0}")]
+    ClusterValidation(String),
+}
+
+/// Validates that all tables with cluster_name reference clusters defined in the config
+fn validate_cluster_references(project: &Project, plan: &InfraPlan) -> Result<(), ValidationError> {
+    let defined_clusters = project.clickhouse_config.clusters.as_ref();
+
+    // Get all cluster names from the defined clusters
+    let cluster_names: Option<Vec<String>> =
+        defined_clusters.map(|clusters| clusters.iter().map(|c| c.name.clone()).collect());
```

Comment:
Nit (suggestion no need to actually do this): Representing this state as option<vec> is a bit hard to follow but instead I would suggest leveraging the empty vec state as the none state and then adjusting the match below to be a set of if statements that each run through the validations. I find that match pretty hard to follow, I'd like to see a `if names.contains(...) { continue }` since I think that reads super clear to me.